module/support/logging/DataLogging/src/DataLogging.cpp

```diff
@@ -46,9 +45,12 @@ namespace module::support::logging {
             uint32_t size = data.data.size() + sizeof(data.hash) + sizeof(timestamp_us);
 
             // If the file isn't open, or writing this message will exceed our max size, make a new file
-            if (!output_file.is_open()
-                || (bytes_written + size + 3 + sizeof(timestamp_us) + sizeof(data.hash)) >= config.output.split_size) {
-                output_file.close();
+            if (!encoder || !encoder->is_open()
+                || (encoder->get_bytes_written() + size + 3 + sizeof(timestamp_us) + sizeof(data.hash))
+                       >= config.output.split_size) {
+                if (encoder) {
+                    encoder->close();
+                }
```

Comment:
Seems redundant
```suggestion
            if (!encoder || !encoder->is_open()
                || (encoder->get_bytes_written() + size + 3 + sizeof(timestamp_us) + sizeof(data.hash))
                       >= config.output.split_size) {
                    encoder->close();
```

Reply:
As `encoder` is a `unique_ptr` `!encoder` will be true if `encoder` points to null, this means that `encoder->close()` will deference a null pointer in this case.

Reply:
`encoder` is a pointer because it has a `ofstream` from zstr. I've opened a [pr](https://github.com/mateidavid/zstr/issues/28) a while ago.